### PR TITLE
util/rowcodec: migrate test-infra to testify

### LINKED ISSUE #26413: migrate test-infra to testify for `util/rowcodec` pkg

### PR DESCRIPTION
**Pull request by UBarney**
### What problem does this PR solve?

Issue Number: close #26413

### Release note 

```release-note
None
```

### CODE REVIEW

File: util/rowcodec/rowcodec_test.go [hunk 1]
```diff
@@ -81,35 +75,36 @@ func (s *testSuite) TestEncodeLargeSmallReuseBug(c *C) {
...
 	m, err := bDecoder.DecodeToDatumMap(b, nil)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	v := m[smallColID]
-	c.Assert(v.GetInt64(), Equals, int64(2))
+	require.Equal(t, int64(2), v.GetInt64())
 }
 
-func (s *testSuite) TestDecodeRowWithHandle(c *C) {
+func TestDecodeRowWithHandle(t *testing.T) {
```
Comment (tisonkun) [inline util/rowcodec/rowcodec_test.go:83]: You may try to run different test data with subtests.

File: util/rowcodec/rowcodec_test.go [hunk 2]
```diff
@@ -519,21 +520,23 @@ func (s *testSuite) TestTypesNewRowCodec(c *C) {
 	}
 
 	// test small
-	encodeAndDecode(c, testData)
+	encodeAndDecode(t, testData)
```
Comment (tisonkun) [inline util/rowcodec/rowcodec_test.go:523]: ditto

File: util/rowcodec/rowcodec_test.go [hunk 3]
```diff
@@ -657,11 +659,13 @@ func (s *testSuite) TestNilAndDefault(c *C) {
 			false,
 		},
 	}
-	encodeAndDecode(c, dtNilData)
+	encodeAndDecode(t, dtNilData)
```
Comment (tisonkun) [inline util/rowcodec/rowcodec_test.go:662]: ditto

File: util/rowcodec/rowcodec_test.go [hunk 4]
```diff
@@ -716,10 +720,12 @@ func (s *testSuite) TestVarintCompatibility(c *C) {
 			false,
 		},
 	}
-	encodeAndDecodeByte(c, testDataValue)
+	encodeAndDecodeByte(t, testDataValue)
```
Comment (tisonkun) [inline util/rowcodec/rowcodec_test.go:723]: ditto

**Review by tisonkun — COMMENTED** [review]
Generally looks good. Suggestions on tests organizing.

File: util/rowcodec/rowcodec_test.go (hunk 1 above)
Comment (tisonkun) [reply util/rowcodec/rowcodec_test.go:83]: see also https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks

Comment (tisonkun) [reply util/rowcodec/rowcodec_test.go:83]: also `TestIsNormalizedDiskNameAllowed` as an example/

File: util/rowcodec/rowcodec_test.go (hunk 3 above)
Comment (tisonkun) [reply util/rowcodec/rowcodec_test.go:662]: make sense

File: util/rowcodec/rowcodec_test.go (hunk 1 above)
Comment (UBarney) [reply util/rowcodec/rowcodec_test.go:88]: done

File: util/rowcodec/rowcodec_test.go (hunk 2 above)
Comment (UBarney) [reply util/rowcodec/rowcodec_test.go]: done

File: util/rowcodec/rowcodec_test.go (hunk 3 above)
Comment (UBarney) [reply util/rowcodec/rowcodec_test.go]: done

File: util/rowcodec/rowcodec_test.go (hunk 4 above)
Comment (UBarney) [reply util/rowcodec/rowcodec_test.go]: inlined

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
/cc @xhebox @tiancaiamao @mmyj 

PTAL.

It seems CI fails due to jenkins failure.

File: util/rowcodec/rowcodec_test.go [hunk 5]
```diff
@@ -518,128 +450,109 @@ func (s *testSuite) TestTypesNewRowCodec(c *C) {
...
+				} else {
+					require.Equal(t, td.bt, d)
+				}
+			}
+		})
 	}
-	dtNilData := []testData{
+}
+
+func TestNilAndDefault(t *testing.T) {
```
Comment (xhebox) [inline util/rowcodec/rowcodec_test.go:552]: Maybe add this test as a subtest into `TestTypesNewRowCodec`. The diff of this PR is hard to review. Inline expands the code too much.

In such case, I would like to see reusing as much as possible, e.g. merging two tests of `encodeAndDecode` into one.

File: util/rowcodec/rowcodec_test.go [hunk 6]
```diff
@@ -30,16 +29,9 @@ import (
...
 
-func TestT(t *testing.T) {
-	TestingT(t)
-}
-
-var _ = Suite(&testSuite{})
-
-type testSuite struct{}
-
 type testData struct {
```
Comment (UBarney) [inline util/rowcodec/rowcodec_test.go:35]: 1. what's diffierence between `dt` and `bt`
2. Does `testData.def` mean default value ?

**Review by UBarney — COMMENTED** [review]
hi @tisonkun, I have questions. Please answer when you have time

File: util/rowcodec/rowcodec_test.go (hunk 6 above)
Comment (tisonkun) [reply util/rowcodec/rowcodec_test.go:35]: I'd like to refer to the original author @lysu . Any idea on this snippet of code?

Comment (lysu) [reply util/rowcodec/rowcodec_test.go:35]: 1. `dt` should be "input" and `bt` should be "output"..sorry for use such short name 😭 

for example, some type like duration input will decode as flatten type -- `int` in this rowcodec level(but will unflatten later in little upper level

https://github.com/pingcap/tidb/blob/da5574a0678e809db3aeea2c8a250ae3011676ea/util/rowcodec/rowcodec_test.go#L426-L427

2. yes `def` means default value and only this case use it
3. 
https://github.com/pingcap/tidb/blob/da5574a0678e809db3aeea2c8a250ae3011676ea/util/rowcodec/rowcodec_test.go#L656

Comment (UBarney) [reply util/rowcodec/rowcodec_test.go:35]: Got it; Thanks

Comment (tisonkun) [reply util/rowcodec/rowcodec_test.go:35]: @UBarney we can rename the variable to fix the readability in this pass, though - by pushing a dedicated commit.

Comment (UBarney) [reply util/rowcodec/rowcodec_test.go:36]: Ok, I'll do it

Comment (UBarney) [reply util/rowcodec/rowcodec_test.go:36]: how about we change `testData` to 
```
type testData struct {
	id           int64
	ft           *types.FieldType
	inputDatum   types.Datum
	expected     types.Datum
	defaultDatum *types.Datum
	handle       bool
}

```

Comment (tisonkun) [reply util/rowcodec/rowcodec_test.go:36]: I'd prefer

```go
type testData struct {
	id           int64
	ft           *types.FieldType
	input   types.Datum
	output     types.Datum
	default *types.Datum
	handle       bool
}
```

we don't have to repeat the field type and keep input and output symmetric. An output from testdata is expected data obviously IMO.

**Comment by tisonkun** [comment]
Please fix the wrong license header, you may try to use license header from other files that pass the check.

**Comment by tiancaiamao** [comment]
If it's ready for review, please remove the Draft @UBarney

**Comment by UBarney** [comment]
Sorry, I have been busy recently. There is no time to change this  pr. Close it for now

**Comment by tisonkun** [comment]
@UBarney ok. I'll take it over.

**Pull request closed without merging**